### PR TITLE
refactor(header component): header component is ready for test

### PR DESCRIPTION
**Pull request by DiLiUA**
- reworked markup for Header component (responsive)
  - creaded short-user-info component
  - used short-user-info component in event-info component.
  - changed User model;
  - rework Auth service

### CODE REVIEW

File: src/header/header.component.html [hunk 1]
```diff
@@ -1,104 +1,141 @@
-<div class="header-container">
-  <div class="container">
-    <div class="title-container">
-      <img src="/assets/img/logo.png" alt="Logo icon" (click)="goToHomePage()">
+<header id="header">
```
Comment (dpereverzev) [inline src/header/header.component.html]: id

File: src/header/header.component.html [hunk 2]
```diff
@@ -1,104 +1,141 @@
...
-              (click)="loginModal()">Login
+    <div class="header-right-container" [class.show]="isShowSideMenu">
+      <button type="button" class="menu-icon" (click)="showMenu()">
+        <i class="fa fa-bars" aria-hidden="true"></i>
       </button>
+      <div class="custom-overlay" (click)="hideMenu()"></div>
+      <div class="side-menu-container">
+        <nav class="menu-container">
+          <ul class="main-menu">
+            <li class="main-menu-item"><a href="#" routerLink="/artists" routerLinkActive="active" (click)="hideMenu()">Artists</a></li>
```
Comment (dpereverzev) [inline src/header/header.component.html]: you dont need href='#'

File: src/header/header.component.html [hunk 3]
```diff
@@ -1,104 +1,141 @@
...
-        </ul>
+        <ng-template #headerUserContainer>
+          <div class="current-user-container">
+            <div dropdown>
+              <div class="current-user-dropdown" dropdownToggle>
+                <app-short-user-info [info]="shortUserInfo" [isLocation]="false"></app-short-user-info>
+              </div>
+              <ul *dropdownMenu role="menu" aria-labelledby="single-button" class="custom-dropdown-menu dropdown-menu dropdown-menu-right">
+                <li role="menuitem"><a class="dropdown-item" href="#" routerLink="/edit-profile" routerLinkActive="active" (click)="hideMenu()">My account</a></li>
+                <li role="menuitem"><a class="dropdown-item" href="#" routerLink="/my-events" routerLinkActive="active" (click)="hideMenu()">My events</a></li>
```
Comment (dpereverzev) [inline src/header/header.component.html]: href='#'

File: src/header/header.component.html [hunk 4]
```diff
@@ -1,104 +1,141 @@
...
-            Let's Rock!
-          </button>
+
+        <div class="fields-container">
+          <div class="form-group">
+            <input type="email"
+                   placeholder="Email"
+                   class="form-control"
+                   name="loginEmail"
+                   pattern="\S+@\S+\.\S+"
```
Comment (dpereverzev) [inline src/header/header.component.html]: i'm not sure that this is the best choice to detect email

File: src/header/header.component.scss [hunk 5]
```diff
@@ -0,0 +1,220 @@
+@import "../variables.scss";
+
+#header {
+  display: block;
```
Comment (dpereverzev) [inline src/header/header.component.scss]: display: block; useless with position:fixed

File: src/header/header.component.scss [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+  width: 100%;
+  height: 75px;
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 1001;
+  background: white;
```
Comment (dpereverzev) [inline src/header/header.component.scss]: hex

File: src/header/header.component.scss [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+.current-user-dropdown {
+  cursor: pointer;
+}
+
+.modal-header h4 {
+	color: #1c2e36;
+	font-size: 16px;
+	font-weight: 400;
```
Comment (dpereverzev) [inline src/header/header.component.scss]: tabs

File: src/header/header.component.scss [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+  line-height: 1;
+  cursor: pointer;
+}
+
+.btn-container {
+  display: flex;
+  justify-content: center;
+}
+
+@media screen and (max-width: 960px) {
```
Comment (dpereverzev) [inline src/header/header.component.scss]: use mixin

File: src/header/header.component.scss [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+  position: relative;
+  padding: 5px 0 5px 10px;
+}
+
+.current-user-dropdown {
+  cursor: pointer;
+}
+
+.modal-header h4 {
+	color: #1c2e36;
```
Comment (dpereverzev) [inline src/header/header.component.scss]: use variable

**Review by dpereverzev — CHANGES_REQUESTED** [review]

File: src/auth/auth.service.ts [hunk 10]
```diff
@@ -159,16 +162,32 @@ export class AuthService {
...
+  authenticated(): boolean {
     // Check if there's an unexpired JWT
     // It searches for an item in localStorage with key == 'id_token' by default
     return tokenNotExpired();
   }
+
+  setUserProfile(profileJson: string): void {
+    try {
+      const profile = JSON.parse(profileJson);
+      if (profile) {
```
Comment (Ugmaxie) [inline src/auth/auth.service.ts]: invert

File: src/header/header.component.ts [hunk 11]
```diff
@@ -1,59 +1,105 @@
...
+
+      });
   }
 
-  ngOnInit() {
+  getUser(): void {
     try {
-      this.userProfile = new User(JSON.parse(this.localStorageService.getItem('profile')));
+      const profile = JSON.parse(this.localStorageService.getItem('profile'));
+      if (profile) {
```
Comment (Ugmaxie) [inline src/header/header.component.ts]: invert

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: src/header/header.component.scss [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+    width: 100%;
+  }
+}
+
+.header-right-container {
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  height: 100%;
+  width: 100%;
```
Comment (Ugmaxie) [inline src/header/header.component.scss]: width
width

File: src/header/header.component.ts [hunk 13]
```diff
@@ -1,59 +1,104 @@
...
+    this.getUser();
 
-  constructor(private auth: AuthService,
-                     private localStorageService: LocalStorageService,
-                     private router: Router) {
+    this.userProfileSubscription = this.auth.userProfileEmitter
+      .subscribe((user: User) => {
+        this.userProfile = user;
+
+        if (user) {
```
Comment (Ugmaxie) [inline src/header/header.component.ts]: invert

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**